modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -2385,12 +2388,30 @@ private void onCacheStarted(GridCacheContext cacheCtx) throws IgniteCheckedExcep
                 ", mode=" + cfg.getCacheMode() +
                 ", atomicity=" + cfg.getAtomicityMode() +
                 ", backups=" + cfg.getBackups() +
-                ", mvcc=" + cacheCtx.mvccEnabled() + ']');
+                ", mvcc=" + cacheCtx.mvccEnabled() +
+                (cacheCtx.expiry() != null ? ", " + buildExpirePolicyInfo(cacheCtx) : "") + ']');
         }
 
         return cacheCtx;
     }
 
+    /**
+     * Get formatted string with expire policy info.
+     *
+     * @param cacheCtx - grid cache context.
+     * @return formatted expire policy info.
+     */
+    private String buildExpirePolicyInfo(GridCacheContext cacheCtx) {
+        ExpiryPolicy expPlc = cacheCtx.expiry();
+
+        Duration dur = expPlc.getExpiryForCreation() != null ? expPlc.getExpiryForCreation() :
```

Comment:
Expiry policy can be null, FYI

Reply:
Sorry, I suppose issue is not here.
You should not log expiry policy like here.
Expiry policy contains 3 fields and all of them can be used.

So you must log all durations of expiry policy. I suggests log -1 if duration is null

Reply:
Thin clients use `org.apache.ignite.internal.processors.platform.cache.expiry.PlatformExpiryPolicy`
It contains field `org.apache.ignite.internal.processors.platform.cache.expiry.PlatformExpiryPolicy#DUR_UNCHANGED`
If thin client pass it, thin client handler convert it to `null`. In tests one of parameter is `ExpiryPolicy()` with all fields `DUR_UNCHANGED`. And this case cause NPE in your code